api/envoy/type/matcher/v3/matcher.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.type.matcher.v3;
+
+import "envoy/type/matcher/v3/header.proto";
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.type.matcher.v3";
+option java_outer_classname = "MatcherProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// TODO(shivanshu21): Once HeaderMatcher is deprecated in favor of StringMatcher,
```

Comment:
Same comment as in last PR. The comments here should indicate what this file is replacing/deprecating (and the original file should be updated to explain where the future version lives.